src/Servers/Kestrel/Core/src/Internal/Http/Http1ContentLengthMessageBody.cs

```diff
@@ -213,17 +214,23 @@ public override void AdvanceTo(SequencePosition consumed, SequencePosition exami
 
             _isReading = false;
 
+        // The current body is read, though there might be more bytes to read on the stream with pipelining
             if (_readCompleted)
             {
-                // If the old stored _readResult was canceled, it's already been observed. Do not store a canceled read result permanently.
-                _readResult = new ReadResult(_readResult.Buffer.Slice(consumed, _readResult.Buffer.End), isCanceled: false, isCompleted: true);
-
-                if (!_finalAdvanceCalled && _readResult.Buffer.Length == 0)
+            if (!_finalAdvanceCalled && consumed.Equals(_readResult.Buffer.End))
```

Comment:
I'm impressed this backported without merge conflicts, even if you do have to go fix all the spacing.